imnodes.cpp

```diff
@@ -705,18 +731,34 @@ void BoxSelectorUpdateSelection(ImNodesEditorContext& editor, ImRect box_rect)
     }
 }
 
+ImVec2 SnapOriginToGrid(ImVec2 origin)
+{
+    if (GImNodes->Style.Flags & ImNodesStyleFlags_GridSnapping)
+    {
+        const float spacing = GImNodes->Style.GridSpacing;
+        const float spacing2 = spacing / 2.0f;
+        float modx = fmodf(fabsf(origin.x) + spacing2, spacing) - spacing2;
+        float mody = fmodf(fabsf(origin.y) + spacing2, spacing) - spacing2;
+        origin.x += (origin.x < 0.f) ? modx : -modx;
+        origin.y += (origin.y < 0.f) ? mody : -mody;
```

Comment:
I think this can be simplified, since the remainder retains the sign of the dividend:

```cpp
        const float spacing = GImNodes->Style.GridSpacing;
        origin.y -= fmodf(origin.y, spacing);
        origin.x -= fmodf(origin.x, spacing);
```

Reply:
It actually cannot be simplified in that way, as this code is responsible for 'rounding' the snap to the next major gridline if the node's origin is beyond the center of the grid cell (so the `spacing2` term does have to be distributed over the equation and the sign applied in some way). However, I believe I can simplify it in a different way.